### PR TITLE
Fix performance issues

### PR DESCRIPTION
**Pull request by anonymoose2**
## Proposed changes
- Do not needlessly reevaluate `User.recommended_activities `
- Do not load announcementrequest.user unnecessarily
- Consider all waitlists to be empty if `ENABLE_WAITLIST` is `False`
- Make `ENABLE_WAITLIST` default `False` and add warning about performance issues

## Brief description of rationale
See #958

### CODE REVIEW

File: intranet/apps/announcements/models.py [hunk 1]
```diff
@@ -167,7 +167,7 @@ def is_visible_requester(self, user):
 
     def is_visible_submitter(self, user):
         try:
-            return (self.announcementrequest and user == self.announcementrequest.user) or self.user == user
+            return (self.announcementrequest and user.id == self.announcementrequest.user_id) or self.user_id == user.id
```
Comment (theo-o) [inline intranet/apps/announcements/models.py]: Where is `user_id` defined?

Comment (anonymoose2) [reply intranet/apps/announcements/models.py]: Django implicitly generates `_id` fields for `ForeignKey`s.

Comment (anonymoose2) [reply intranet/apps/announcements/models.py]: Assuming that the primary key is `id`, of course -- which it is for our `User` model.

File: intranet/apps/eighth/serializers.py [hunk 2]
```diff
@@ -147,12 +148,16 @@ def process_scheduled_activity(
             "title": scheduled_activity.title,
             "comments": scheduled_activity.comments,
             "display_text": "",
-            "waitlist_count": EighthWaitlist.objects.filter(scheduled_activity_id=scheduled_activity.id).count(),
+            "waitlist_count": (EighthWaitlist.objects.filter(scheduled_activity_id=scheduled_activity.id).count() if settings.ENABLE_WAITLIST else 0),
```
[the remark this answers is not in the document]
Comment (anonymoose2) [reply intranet/apps/eighth/serializers.py]: Done.

File: intranet/apps/eighth/serializers.py [hunk 3]
```diff
@@ -147,12 +148,17 @@ def process_scheduled_activity(
             "title": scheduled_activity.title,
             "comments": scheduled_activity.comments,
             "display_text": "",
-            "waitlist_count": EighthWaitlist.objects.filter(scheduled_activity_id=scheduled_activity.id).count(),
+            # TODO: Make waitlists more efficient
```
Comment (theo-o) [inline intranet/apps/eighth/serializers.py]:
```suggestion
            # TODO: Make waitlists more efficient or remove functionality completely
```

**Pull request merged**